File: 2540/MinimumCommonValue.ts
/*
https://leetcode.com/problems/minimum-common-value/description/?envType=daily-question&envId=2024-03-09
*/

function getCommon(nums1: number[], nums2: number[]): number {
    let mp:Map<number, number>=new Map();
    let sz:number=nums1.length, sz2:number=nums2.length;
    for(let i:number=0; i<sz; ++i){
        if(mp.has(nums1[i])){
            mp.set(nums1[i], mp.get(nums1[i])+1);
        } else mp.set(nums1[i], 1);
    }
    for(let i:number=0; i<sz2; ++i){
        if(mp.has(nums2[i])) return nums2[i];
    }
    return -1;
};

File: 2540/MinimumCommonValue.cs
/*
https://leetcode.com/problems/minimum-common-value/description/?envType=daily-question&envId=2024-03-09
*/

public class Solution {
    public int GetCommon(int[] nums1, int[] nums2) {
        Dictionary<int, int>dict=new Dictionary<int, int>();
        int sz1=nums1.Length, sz2=nums2.Length;
        for(int i=0; i<sz1; ++i){
            if(dict.ContainsKey(nums1[i])){
                dict[nums1[i]]+=1;
            } else dict.Add(nums1[i], 1);
        }
        for(int i=0; i<sz2; ++i){
            if(dict.ContainsKey(nums2[i])) return nums2[i];
        }
        return -1;
    }
}